SmartStorePerformance/js/app.js

```diff
@@ -0,0 +1,304 @@
+                        mode="dropdown"
+                        selectedValue={this.state.buildSpec}
+                        onValueChange={this.updateBuildSpec.bind(this)}
+                        >
+                        <Picker.Item label="All Query Spec" value="all" />
+                        <Picker.Item label="Exact Query Spec" value="exact" />
+                        <Picker.Item label="Range Query Spec" value="range" />
+                        <Picker.Item label="Like Query Spec" value="like" />
+                        <Picker.Item label="Match Query Spec" value="match" />
+                    </Picker>
+
+                    {this.state.buildSpec == 'exact' &&
+                    <TextInput
+                        style={{height: 40, borderColor: 'gray', borderWidth: 1, padding: 10}}
+                        onChangeText={(exactKey) => this.setState({exactKey})}
+                        value={this.state.exactKey}
+                    />}
+                    {this.state.buildSpec == 'range' &&
+                    <TextInput
+                        style={{height: 40, borderColor: 'gray', borderWidth: 1, padding: 10}}
+                        onChangeText={(beginKey) => this.setState({beginKey})}
+                        value={this.state.beginKey}
+                    />
+                    }
+                    {this.state.buildSpec == 'range' &&
+                    <TextInput
+                        style={{height: 40, borderColor: 'gray', borderWidth: 1, padding: 10}}
+                        onChangeText={(endKey) => this.setState({endKey})}
+                        value={this.state.endKey}
+                    />
+                    }
+                    {this.state.buildSpec == 'like' &&
+                    <TextInput
+                        style={{height: 40, borderColor: 'grey', borderWidth: 1}}
+                        onChangeText={(likeKey) => this.setState({likeKey})}
+                        value={this.state.likeKey}
+                    />}
+                    {this.state.buildSpec == 'match' &&
+                     <TextInput
+                         style={{height: 40, borderColor: 'gray', borderWidth: 1}}
+                         onChangeText={(matchKey) => this.setState({matchKey})}
+                         value={this.state.matchKey}
+                     />}
+
+                    <Slider
+                        value={this.state.pageSize}
+                        onValueChange={(value) => this.setState({pageSize: value})}
+                        minimumValue={1}
+                        maximumValue={100}
+                        step={1}
+                    />
+                    <Text>Page Size: {this.state.pageSize}</Text>
+
+                    <Slider
+                        value={this.state.recordLimit}
+                        onValueChange={(value) => this.setState({recordLimit: value})}
+                        minimumValue={0}
+                        maximumValue={1000}
```

Comment:
Everything is running really fast. Could we make that value larger? Maybe 50000?

Reply:
@wmathurin Yes, but should we be concerned due to it being stored locally?  The json file that has this data is currently 1.4Mb with 1000 records.  Are we okay multiplying that by 50?

Reply:
I would leave the file alone.
In `populateSoups` you could create extra records in memory based on the ones found in the file.
```javascript
const x1data = require('./X1_Custom_Perf.json').records;
for (var j=0; j<50; j++) {
  const derivedData = deriveData(x1data, j);
  upsertSoupEntries(false, selectSoup, derivedData)
    .catch((error) => {
      console.log("Populating soup '" + selectSoup + "' failed with error: " + error);
    });
}
```
With 
```javascript
function padZ(s, n) {
return  ('0000000000' + s).slice(-n)
}

function deriveData(originalData, j) {
  var derivedData = [];
  for (var i=0; i<originalData.length; i++) {
    var record = originalData[i];
    record.Id = record.Id.replace("000000", padZ("" + j, 6)) // e.g. will replace aJUx000000000DWGAY with aJUx000013000DWGAY if j = 13
    record.name = record.name + "_" + i;
    // more mutations
    derivedData.push[record];
  }
  return derivedData;
}
```